### PR TITLE
feature: ordinal scale support specified

### PR DESCRIPTION
**Pull request by xuefei1313**
[[中文版模板 / Chinese template](https://github.com/VisActor/Vutil/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Chore
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     ordinal scale support specified      |
| 🇨🇳 Chinese |   离散scale支持 specified       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vscale/src/ordinal-scale.ts [hunk 1]
```diff
@@ -9,6 +9,24 @@ export class OrdinalScale extends BaseScale implements IBaseScale {
   protected _index: Map<string, number>;
   protected _domain: Array<number>;
   protected _ordinalRange: Array<number>;
+  /** specified: support scale to return specific value on special input value */
+  protected _specified: Record<string, unknown>;
+  specified(): Record<string, unknown>;
+  specified(_: Record<string, unknown>): this;
+  specified(_?: Record<string, unknown>): this | Record<string, unknown> {
+    if (!_) {
+      return { ...this._specified };
```
Comment (xile611) [inline packages/vscale/src/ordinal-scale.ts:18]: `{... }` could be Performance-consuming, use Object.assign replaced

File: packages/vscale/src/ordinal-scale.ts [hunk 2]
```diff
@@ -24,6 +42,10 @@ export class OrdinalScale extends BaseScale implements IBaseScale {
   }
 
   scale(d: any): any {
+    const special = this._getSpecifiedValue(d);
+    if (special) {
```
Comment (xile611) [inline packages/vscale/src/ordinal-scale.ts:46]: how about zero or blank space is a specified value ?

Comment (xuefei1313) [reply packages/vscale/src/ordinal-scale.ts:46]: o ~yes , i will fix this

**Review by xile611 — APPROVED** [review]
LGTM

**Pull request merged**